logdevice/common/test/RandomNodeSelectorTest.cpp

```diff
@@ -175,23 +176,44 @@ inline NodeSourceSet genRandomSet(node_index_t min,
 TEST(RandomNodeSelector, SourceSelectBasicTest) {
   {
     auto result = RandomNodeSelector::select(
-        {1, 2, 3, 4, 5}, {1}, {2}, {3}, 2, 1, nullptr);
+        {1, 2, 3, 4, 5}, {1}, {2}, {3}, 2, 1, folly::none, nullptr);
     NodeSourceSet expected{3, 4, 5};
     EXPECT_EQ(expected, result);
   }
   {
     auto result = RandomNodeSelector::select(
-        {1, 2, 3, 4, 5}, {1}, {2, 3}, {}, 2, 1, nullptr);
+        {1, 2, 3, 4, 5}, {1}, {2, 3}, {}, 2, 1, folly::none, nullptr);
     NodeSourceSet expected{4, 5};
     EXPECT_EQ(expected, result);
   }
   {
     auto result = RandomNodeSelector::select(
-        {1, 2, 3, 4, 5}, {1, 4}, {2, 3}, {}, 2, 0, nullptr);
+        {1, 2, 3, 4, 5}, {1, 4}, {2, 3}, {}, 2, 0, folly::none, nullptr);
     EXPECT_TRUE(result.empty());
   }
 }
 
+TEST(RandomNodeSelector, SameOrderTest) {
+  for (int i = 0; i < 100; i++) {
+    srand(time(0));
+    folly::Optional<u_int32_t> seed = std::rand();
```

Comment:
Avoid using `std::rand()` and use `folly::Random::rand32()`. You don't need to seed it as well.